### PR TITLE
Update sky130hd klayout technology file

### PR DESCRIPTION
**Pull request by antonblanchard**
This fixes a number of issues in the sky130hd klayout technology file.

### CODE REVIEW

**Comment by antonblanchard** [comment]
An example of the issue:

bad

And with this applied:

good

File: flow/platforms/sky130hd/sky130hd.lyt [hunk 1]
```diff
@@ -74,6 +94,16 @@
...
+  <mag>
+   <lambda>1</lambda>
+   <dbu>0.001</dbu>
+   <layer-map>layer_map()</layer-map>
+   <create-other-layers>true</create-other-layers>
+   <keep-layer-names>false</keep-layer-names>
+   <merge>true</merge>
+   <lib-paths>
+   </lib-paths>
+  </mag>
```
Comment (rovinski) [inline flow/platforms/sky130hd/sky130hd.lyt:106]: We don't use Magic in ORFS but I'll assume this is fine.

File: flow/platforms/sky130hd/sky130hd.lyt [hunk 2]
```diff
@@ -101,7 +132,72 @@
...
+# Example:
+#   var thick_m1 = true
+#   if thickm1
+#     1: 0.5 1.5
+#   else
+#     1: 0.5 1.2
+#   end
+
+</src>
+ </d25>
```
Comment (rovinski) [inline flow/platforms/sky130hd/sky130hd.lyt]: No need for this; please remove.

File: flow/platforms/sky130hd/sky130hd.lyt [hunk 3]
```diff
@@ -101,7 +132,72 @@
...
  <connectivity>
+  <connection>li1,67/44,met1</connection>
+  <connection>met1,68/44,met2</connection>
+  <connection>met2,69/44,met3</connection>
+  <connection>met3,70/44,met4</connection>
+  <connection>met4,71/44,met5</connection>
```
Comment (rovinski) [inline flow/platforms/sky130hd/sky130hd.lyt]: Can you please add vias as symbols instead of GDS layer/data?

Comment (rovinski) [reply flow/platforms/sky130hd/sky130hd.lyt]: Also do you know if via pin layers count towards connectivity? If so, please add them.

**Review by rovinski — CHANGES_REQUESTED** [review]

File: flow/platforms/sky130hd/sky130hd.lyt (hunk 2 above)
Comment (antonblanchard) [reply flow/platforms/sky130hd/sky130hd.lyt]: Ok, removed.

File: flow/platforms/sky130hd/sky130hd.lyt (hunk 3 above)
Comment (antonblanchard) [reply flow/platforms/sky130hd/sky130hd.lyt]: Added the vias as symbols. I'm not sure about the via pin layers, I can't find any reference to the layer/data for them in either the magic tech file or sky130 PDK documentation. Also somewhat suspicious is the datatype is not consistent between layers:

```
via.pin - 68/58
via2.pin - 69/58
via3.pin - 70/48
via4.pin - 71/48
```

Comment (antonblanchard) [reply flow/platforms/sky130hd/sky130hd.lyt]:
```
<symbols>li1='67/20+67/5+67/16'</symbols>
<symbols>met1='68/20+68/5+68/16'</symbols>
<symbols>met2='69/20+69/5+69/16'</symbols>
<symbols>met3='70/20+70/5+70/16'</symbols>
<symbols>met4='71/20+71/5+71/16'</symbols>
<symbols>met5='72/20-72/15+72/5+72/16'</symbols>
```

I removed the labels (datatype 5), seems wrong to be including them here. Another thing - why do we subtract datatype 5 (short) only from met5? From reading the PDK these are some sort of resistor for metal layers, and if so shouldn't we subtract it from all metal layers?

Comment (rovinski) [reply flow/platforms/sky130hd/sky130hd.lyt]: > I'm not sure about the via pin layers, I can't find any reference to the layer/data for them in either the magic tech file or sky130 PDK documentation.

Yes, that's strange. Let's ignore it for now.

> Another thing - why do we subtract datatype 5 (short) only from met5

I'm not sure I follow - there was no connectivity in the master branch version. Your PR is adding it. I'm guessing you're copying this from elsewhere, in which case I don't know what it's supposed to mean or if it's a mistake.

**Review by rovinski — APPROVED** [review]

**Pull request merged**